### PR TITLE
bpo-31307: Allow use of bytes object as a single argument in ConfigParser.read

### PR DESCRIPTION
**Pull request by vxgmichel**
Fix issue [bpo-31307](http://bugs.python.org/issue31307) and [bpo-29627](http://bugs.python.org/issue29627).

https://bugs.python.org/issue31307

### CODE REVIEW

**Review by merwok — APPROVED** [review]

File: Lib/test/test_configparser.py [hunk 1]
```diff
@@ -739,6 +740,18 @@ def test_read_returns_file_list(self):
         cf = self.newconfig()
         parsed_files = cf.read([])
         self.assertEqual(parsed_files, [])
+        # check when passing an existing bytestring path
```
Comment (berkerpeksag) [inline Lib/test/test_configparser.py]: `test_read_returns_file_list` is already too long, could you move it a separate test case?

File: Misc/NEWS.d/next/Library/2017-09-07-12-50-28.bpo-31307.AVBiNY.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Allow use of bytes objects for arguments to configparser.ConfigParser.read.
```
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Library/2017-09-07-12-50-28.bpo-31307.AVBiNY.rst]: Please add "Patch by Vincent Michel.".

Comment (berkerpeksag) [inline Misc/NEWS.d/next/Library/2017-09-07-12-50-28.bpo-31307.AVBiNY.rst]: :meth:\`configparser.ConfigParser.read\`

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by vxgmichel** [comment]
@berkerpeksag Thanks for the review. I have made the requested changes; please review again.

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

**Comment by ambv** [comment]
Thanks! ✨ 🍰 ✨

**Pull request merged**